core/dbt/tests/fixtures/project.py

```diff
@@ -118,16 +128,22 @@ def project_config_update():
     return {}
 
 
+@pytest.fixture
+def model_path():
```

Comment:
This part is needed in some tests that have more complex file structure

Reply:
The model dictionaries can be nested. i.e.:
`{ "my_model.sql": '....', "subdir": { "another_model.sql": "..."}}`

Reply:
Yea, but this is used to specify model files to use for a run. previous integration tests supported it [here](https://github.com/dbt-labs/dbt-core/blob/5e6be1660eb5672c9bfc3d79aa02cec275de7533/test/integration/base.py#L359)

Reply:
Which tests in the old integration tests? I know that the old integration test framework did that, but it was because they had directories for multiple projects in the same test directory, and we don't need that anymore.

Reply:
This [one ](https://github.com/dbt-labs/dbt-core/blob/5615691c329ca55b5782b8f98f75b36bb3a081d6/tests/functional/schema_tests/test_schema_v2_tests.py#L38). Basically every test class has it's own models

Reply:
I can refactor it to just one models artifact per class if we want to do that.

Reply:
Yes, but we don't need that anymore. We just construct a models fixture with the right files. They did that because every test had to have a separate directory, otherwise tests would interfere with each other.

Reply:
I think one models fixture per class is probably better.

Reply:
@gshank  this is resolved, anything else we would like to change about this PR?